indigo/bulk_creator.py

```diff
@@ -382,19 +381,26 @@ def link_publication_document(self, work, info):
         pub_doc.save()
 
     def link_commencement(self, work, info):
-        # if the work is `commenced_by` something, try linking it
-        # make a task if this fails
-        title = info['commenced_by']
-        work = info['work']
-        commencing_work = self.find_work_by_title(title)
-        if not commencing_work:
-            return self.create_task(work, info, task_type='link-commencement')
-
-        work.commencing_work = commencing_work
-        try:
-            work.save_with_revision(self.user)
-        except ValidationError:
-            self.create_task(work, info, task_type='link-commencement')
+        # if the work has commencement details, try linking it
+        # make a task if a `commenced_by` title is given but not found
+        date = info.get('commencement_date')
+
+        commencing_work = None
+        title = info.get('commenced_by')
+        if title:
+            commencing_work = self.find_work_by_title(title)
+            if not commencing_work:
+                self.create_task(work, info, task_type='link-commencement')
```

Comment:
Do we need to return here?

Reply:
we've changed the logic slightly: 
previously, this was only for linking the commencing work, which is why we had a return here.
now, this is the only place a commencement object is created.
we could change it so that if there's no `commencing_work` and no `date`, we return `here`; but if there's no `commencing_work` and there is a `date`, we create a task here but continue to create the commencement object. 
but if we don't find the commencing_work and there is no date, and we create a shell of a commencement object, that's also not the end of the world? it'll get edited once the task is done, or deleted if the work is marked as uncommenced after all.
lmk if you'd like me to add the extra logic, I don't think it's necessary but if you'd prefer to be cautious we can do it!